mitosheet/src/jupyter/api.tsx

```diff
@@ -503,6 +503,42 @@ export default class MitoAPI {
         return undefined;
     }
 
+    /*
+        Gets a preview of the split text to columns step
+    */
+    async getSplitTextToColumnsPreview(
+        sheetIndex: number,
+        columnID: ColumnID,
+        delimiters: string[],
+    ): Promise<(string | number | boolean)[][] | undefined> {
+
+        const dfPreviewString = await this.send<string>({
+            'event': 'api_call',
+            'type': 'get_split_text_to_columns_preview',
+            'params': {
+                'sheet_index': sheetIndex,
+                'column_id': columnID,
+                'delimiters': delimiters,
+            },
+        }, {})
+
+        if (dfPreviewString !== undefined && dfPreviewString !== '') {
+            const dfPreviewSheetData: SheetData = JSON.parse(dfPreviewString)['dfPreviewSheetData'];
+
+            const dfPreviewArray: (string | number | boolean)[][] = []
```

Comment:
It would be nice if we have a more ergonomic way to get data to the frontend, given that we do this sort of parsing in a few places. 

What would a good type be for this and the unique values? I'm thinking this map:
```
ColumnDataMap: Record<ColumnID => (string | number | boolean)[]>
```

Wanna take a shot at refactoring these two api calls to use this? It would just require writing a new backend and frontend utils (and in return, all this special case code would go away). 

It will be much easier to work with, much cleaner, and much better for maintainability long-term I think!

Reply:
Nice. I made a new util. But the format that is most helpful for these cases is row-organized, instead of column organized, since the transformation that we end up performing in api.tsx is converting the data from column-first, to row-first. So check out the new util `get_row_data_array` in `utils.py`